bin/node/e2e-tests/src/lib.rs

```diff
@@ -0,0 +1,191 @@
+            frame_system::CheckWeight::<Self::Runtime>::new(),
+            pallet_transaction_payment::ChargeTransactionPayment::<Self::Runtime>::from(0),
+        )
+    }
+
+    fn create_client_parts(
+        config: &Configuration,
+    ) -> Result<
+        (
+            Arc<TFullClient<Self::Block, Self::RuntimeApi, Self::Executor>>,
+            Arc<TFullBackend<Self::Block>>,
+            SyncCryptoStorePtr,
+            TaskManager,
+            InherentDataProviders,
+            Option<
+                Box<
+                    dyn ConsensusDataProvider<
+                        Self::Block,
+                        Transaction = sp_api::TransactionFor<
+                            TFullClient<Self::Block, Self::RuntimeApi, Self::Executor>,
+                            Self::Block,
+                        >,
+                    >,
+                >,
+            >,
+            Self::SelectChain,
+            Self::BlockImport,
+        ),
+        sc_service::Error,
+    > {
+        let (client, backend, keystore, task_manager) =
+            new_full_parts::<Self::Block, Self::RuntimeApi, Self::Executor>(config)?;
+        let client = Arc::new(client);
+
+        let inherent_providers = InherentDataProviders::new();
+        let select_chain = sc_consensus::LongestChain::new(backend.clone());
+
+        let (grandpa_block_import, ..) =
+            grandpa::block_import(client.clone(), &(client.clone() as Arc<_>), select_chain.clone())?;
+
+        let (block_import, babe_link) = sc_consensus_babe::block_import(
+            sc_consensus_babe::Config::get_or_compute(&*client)?,
+            grandpa_block_import,
+            client.clone(),
+        )?;
+
+        let consensus_data_provider = BabeConsensusDataProvider::new(
+            client.clone(),
+            keystore.sync_keystore(),
+            &inherent_providers,
+            babe_link.epoch_changes().clone(),
+            vec![(AuthorityId::from(Alice.public()), 1000)],
+        )
+            .expect("failed to create ConsensusDataProvider");
+
+        Ok((
+            client,
+            backend,
+            keystore.sync_keystore(),
+            task_manager,
+            inherent_providers,
+            Some(Box::new(consensus_data_provider)),
+            select_chain,
+            block_import,
+        ))
+    }
+}
+
+/// performs a runtime upgrade given wasm blob and a handle to a node.
+pub fn perform_runtime_upgrade(node: &Node<NodeTemplateChainInfo>) {
+    type SystemCall = frame_system::Call<Runtime>;
+    type SudoCall = pallet_sudo::Call<Runtime>;
+
+    let whales = vec![
+        "5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY", //Alice
+        "5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty", //Bob
+    ]
+        .into_iter()
+        .map(|account| AccountId32::from_str(account).unwrap())
+        .collect::<Vec<_>>();
+
+    let wasm = include_bytes!("./runtime.wasm");
```

Comment:
We should probably not include a compiled binary in the repo. Considering that we are having a wasm-builder-runner is there no way to make this more generic and compile the runtime on the fly? Otherwise this will be out-of-date and test a pretty old-runtime fairly quickly.

Reply:
yeahh, i included it for testing purposes only

Reply:
> yeahh, i included it for testing purposes only
So, it should not be merged?

Then please remove it. I like that you want to show how to use it, but maybe do that in an example rrust-doc then?